addon/-private/system/store.js

```diff
@@ -2713,10 +2713,12 @@ Store = Service.extend({
     if (isNone(resourceIdentifier)) {
       return;
     }
+    assert(`A ${relationship.internalModel.modelName} record was pushed into the store with the value of ${relationship.key} being '${JSON.stringify(resourceIdentifier)}', but ${relationship.key} is a belongsTo relationship so the value must not be an array. You should probably check your data payload or serializer.`, !Array.isArray(resourceIdentifier));
+    assert(
+      `Ember Data expected the data for the ${relationship.key} relationship on a ${relationship.internalModel.toString()} to be in a JSON API format and include an \`id\` and \`type\` property but it found '${JSON.stringify(resourceIdentifier)}'. Please check your serializer and make sure it is serializing the relationship payload into a JSON API format.`,
```

Comment:
Is this after us normalizing the id? I'm a bit worried we could get trolled here by id being 0

Reply:
I believe so but I’ll double check.

Reply:
Good catch, addressed.